apps/web/src/services/dashboards/components/dashboard-detail/DashboardShareModal.vue

```diff
@@ -0,0 +1,298 @@
+import type { PublicDashboardUnshareParameters } from '@/schema/dashboard/public-dashboard/api-verbs/unshare';
+import type { PublicFolderShareParameters } from '@/schema/dashboard/public-folder/api-verbs/share';
+import type { PublicFolderUnshareParameters } from '@/schema/dashboard/public-folder/api-verbs/unshare';
+import { i18n } from '@/translations';
+
+import { useAppContextStore } from '@/store/app-context/app-context-store';
+import { useAllReferenceStore } from '@/store/reference/all-reference-store';
+import type {
+    PublicDashboardReferenceItem,
+    PublicDashboardReferenceMap,
+} from '@/store/reference/public-dashboard-reference-store';
+import type { PublicFolderReferenceItem, PublicFolderReferenceMap } from '@/store/reference/public-folder-reference-store';
+
+import { showErrorMessage, showSuccessMessage } from '@/lib/helper/notice-alert-helper';
+
+import ErrorHandler from '@/common/composables/error/errorHandler';
+import { useProxyValue } from '@/common/composables/proxy-state';
+
+import { gray } from '@/styles/colors';
+
+import { useDashboardDetailInfoStore } from '@/services/dashboards/stores/dashboard-detail-info-store';
+import type { DashboardDataTableItem } from '@/services/dashboards/types/dashboard-folder-type';
+
+
+
+const TABLE_FIELDS = [
+    { name: 'name', label: 'Name' },
+    { name: 'location', label: 'Location' },
+];
+interface Props {
+    visible: boolean;
+    dashboardId: string;
+}
+const props = withDefaults(defineProps<Props>(), {
+    visible: false,
+});
+const emit = defineEmits<{(e: 'update:visible', value: boolean): void;
+}>();
+const appContextStore = useAppContextStore();
+const dashboardDetailStore = useDashboardDetailInfoStore();
+const allReferenceStore = useAllReferenceStore();
+const storeState = reactive({
+    isAdminMode: computed(() => appContextStore.getters.isAdminMode),
+    dashboards: computed<PublicDashboardReferenceMap>(() => allReferenceStore.getters.publicDashboard),
+    folders: computed<PublicFolderReferenceMap>(() => allReferenceStore.getters.publicFolder),
+});
+const state = reactive({
+    loading: false,
+    proxyVisible: useProxyValue<boolean>('visible', props, emit),
+    targetDashboardItem: computed<PublicDashboardReferenceItem|undefined>(() => storeState.dashboards[props.dashboardId]),
+    targetFolderItem: computed<PublicFolderReferenceItem|undefined>(() => {
+        if (!state.hasFolder) return undefined;
+        return storeState.folders[state.targetDashboardItem?.data?.folderId || ''];
+    }),
+    shareToMenuList: computed<MenuItem[]>(() => ([
+        { label: i18n.t('DASHBOARDS.ALL_DASHBOARDS.ALL_WORKSPACES'), name: 'WORKSPACE' },
+        { label: i18n.t('DASHBOARDS.ALL_DASHBOARDS.ALL_PROJECTS'), name: 'PROJECT' },
+    ])),
+    selectedShareTo: 'WORKSPACE' as 'WORKSPACE' | 'PROJECT',
+    needToShare: computed<boolean>(() => !state.targetDashboardItem?.data?.shared),
```

Comment:
Suggestion: **I think** naming `needToShare` is unclear. How about `isUnsharedDashbaord`, `isNotYetShared` or something..? (Just suggestion)